server/app/services/rpc/container_handler.rb

```diff
@@ -32,6 +32,18 @@ def save(data)
       {}
     end
 
+    # @param [String] node_id
+    # @param [Array<String>] ids
+    def cleanup(node_id, ids)
+      node = @grid.host_nodes.find_by(node_id: node_id)
+      if node
+        @grid.containers.where(
+          :host_node_id => node.id,
+          :container_id.in => ids
+        ).delete
```

Comment:
delete vs. destroy?

This won't delete `type: 'volume` containers as the default scope limits only to "normal" containers. I think this should use  `@grid.containers.unscoped....`